akka-projection-jdbc/src/main/scala/akka/projection/jdbc/internal/JdbcOffsetStore.scala

```diff
@@ -60,17 +63,37 @@ private[akka] class JdbcOffsetStore[S <: JdbcSession](
       // init Statement in try-with-resource
       tryWithResource(conn.prepareStatement(settings.dialect.readOffsetQuery)) { stmt =>
 
+        stmt.setFetchSize(settings.fetchSize)
         stmt.setString(1, projectionId.name)
-        stmt.setString(2, projectionId.key)
 
         // init ResultSet in try-with-resource
         tryWithResource(stmt.executeQuery()) { resultSet =>
-          // FIXME: need to iterate over all results to build MultipleOffsets
-          if (resultSet.first()) {
+
+          val buffer = ListBuffer.empty[SingleOffset]
+
+          while (resultSet.next()) {
             val offsetStr = resultSet.getString("OFFSET")
             val manifest = resultSet.getString("MANIFEST")
-            Some(fromStorageRepresentation[Offset](offsetStr, manifest))
-          } else None
+            val mergeable = resultSet.getBoolean("MERGEABLE")
+            val key = resultSet.getString("PROJECTION_KEY")
+
+            val adaptedProjectionId = ProjectionId(projectionId.name, key)
+            val single = SingleOffset(adaptedProjectionId, manifest, offsetStr, mergeable)
+            buffer.addOne(single)
+          }
+
+          if (buffer.isEmpty) None
+          else if (buffer.size == 1) Some(fromStorageRepresentation[Offset, Offset](buffer.head))
+          else if (buffer.forall(_.mergeable)) {
+            Some(
+              fromStorageRepresentation[MergeableOffset[_, _], Offset](MultipleOffsets(buffer.toList))
+                .asInstanceOf[Offset])
+          } else {
+            // if this happen it's a bug, can be a offset store that got corrupt
+            throw new RuntimeException(
+              s"Found more then one offset for [$projectionId], but not all of them are mergeable!")
+          }
```

Comment:
why is this implemented slightly differently than in SlickOffsetStore?

Reply:
hmm, I thought that was fixing a bug, but I was wrong. 

My thought was that it should not be possible to fetch more then one row and not have mergeable records, but that's because I was still thinking on terms of search by `name` and `key` which is not the case anymore here. We search by name only. 

I will fix it. 

I think we can still have an extra protection, for instance: if we see that we have a mix of mergeable and non-mergeable offsets, then the offset is corrupt. We don't want to return `None` like the Slick impl is doing because it will cause projections to start from scratch.

Btw, I was planning to make the Slick one depend on the JDBC that's why I didn't change it to align with this impl.